src/overlays/actors/ovl_En_Fire_Rock/z_en_fire_rock.c

```diff
@@ -55,27 +61,340 @@ static ColliderCylinderInit D_80A12CA0 = {
+                    flamePos.y = this->actor.floorHeight;
+                    flamePos.z = Rand_CenteredFloat(20.0f) + this->actor.world.pos.z;
+                    EffectSsEnFire_SpawnVec3f(globalCtx, &this->actor, &flamePos, 300, 0, 0, -1);
+                }
+                this->actionFunc = EnFireRock_SpawnMoreBrokenPieces;
+                break;
+            default:
+                func_80033260(globalCtx, &this->actor, &this->actor.world.pos, this->actor.shape.shadowScale, 3, 8.0f,
+                              200, 10, 0);
+                Audio_PlaySoundAtPosition(globalCtx, &this->actor.world.pos, 40, NA_SE_EV_EXPLOSION);
+                Actor_Kill(&this->actor);
+                break;
+        }
+    }
+}
+
+/**
+ * After the rock has already hit the ground and started rolling, spawn two more, giving the illusion of breaking into
+ * two pieces.
+ */
+void EnFireRock_SpawnMoreBrokenPieces(EnFireRock* this, GlobalContext* globalCtx) {
+    EnFireRock* spawnedFireRock;
+    s32 nextRockType;
+    s32 i;
+    s32 temp;
+
+    nextRockType = FIRE_ROCK_SPAWNED_FALLING1;
+    switch (this->type) {
+        case FIRE_ROCK_SPAWNED_FALLING1:
+        case FIRE_ROCK_SPAWNED_FALLING2:
+            nextRockType = FIRE_ROCK_BROKEN_PIECE1;
+            break;
+        case FIRE_ROCK_BROKEN_PIECE1:
+            nextRockType = FIRE_ROCK_BROKEN_PIECE2;
+    }
+
+    if (nextRockType != FIRE_ROCK_SPAWNED_FALLING1) {
+        for (i = 0; i < 2; i++) {
+            spawnedFireRock = (EnFireRock*)Actor_Spawn(
+                &globalCtx->actorCtx, globalCtx, ACTOR_EN_FIRE_ROCK, Rand_CenteredFloat(3.0f) + this->actor.world.pos.x,
+                Rand_CenteredFloat(3.0f) + (this->actor.world.pos.y + 10.0f),
+                Rand_CenteredFloat(3.0f) + this->actor.world.pos.z, 0, 0, 0, nextRockType);
+            if (spawnedFireRock != NULL) {
+                spawnedFireRock->actor.world.rot.y = this->actor.world.rot.y;
+                if (i == 0) {
+                    spawnedFireRock->actor.shape.rot.y = this->actor.shape.rot.y;
+                }
+                spawnedFireRock->scale = this->scale - 0.01f;
+            } else {
+                osSyncPrintf(VT_FGCOL(YELLOW) "☆☆☆☆☆ イッパイデッス ☆☆☆☆☆ \n" VT_RST);
+            }
+        }
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_VALVAISA_ROCK);
+    }
+    Actor_Kill(&this->actor);
+}
+
+void FireRock_WaitSpawnRocksFromCeiling(EnFireRock* this, GlobalContext* globalCtx) {
+    EnFireRock* spawnedFireRock;
+
+    if (this->actor.xzDistToPlayer < 200.0f) {
+        if ((this->playerNearby == 0) && (this->timer2 == 0)) {
+            this->timer2 = 30;
+            spawnedFireRock = (EnFireRock*)Actor_Spawn(
+                &globalCtx->actorCtx, globalCtx, ACTOR_EN_FIRE_ROCK, Rand_CenteredFloat(3.0f) + this->actor.world.pos.x,
+                this->actor.world.pos.y + 10.0f, Rand_CenteredFloat(3.0f) + this->actor.world.pos.z, 0, 0, 0,
+                FIRE_ROCK_SPAWNED_FALLING2);
```

Comment:
did the formatter do this? usually it tries to squeeze everything under the parenthesis of the call.
would run it just incase